sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Triggers.scala

```diff
@@ -31,7 +31,7 @@ private object Triggers {
 
   def convert(interval: String): Long = {
     val cal = IntervalUtils.fromString(interval)
-    if (cal.months > 0) {
+    if (cal.months != 0) {
```

Comment:
This seems like another way of converting interval to duration: make sure the `months` field is 0. Shall we also take it into account in the new `getDuration` method?

Reply:
We can change `getDuration()` to:
```scala
  def getDuration(
      interval: CalendarInterval,
      targetUnit: TimeUnit,
      daysPerMonth: Option[Int] = Some(31)): Long = {
    val monthsDuration = daysPerMonth
      .map { days =>
        Math.multiplyExact(days * DateTimeUtils.MICROS_PER_DAY, interval.months)
      }.getOrElse {
        if (interval.months == 0) {
          0L
        } else {
          throw new IllegalArgumentException(s"Doesn't support month or year interval: $interval")
        }
      }
    val result = Math.addExact(interval.microseconds, monthsDuration)
    targetUnit.convert(result, TimeUnit.MICROSECONDS)
  }
```
and call `getDuration(cal, TimeUnit.MILLISECONDS, None)`

Reply:
but I am not sure that this check should be inside of `getDuration()`